### PR TITLE
feat(summary): add taskId to single package task

### PR DESCRIPTION
**Pull request by tknickman**
### Description

`taskId` should always exist.

### CODE REVIEW

**Review by mehulkar — APPROVED** [review]
I don't have a strong objection with this, but:
- `taskId` all over the codebase is `pkg#task`, i'm concerned this exception is confusing
- this duplicates the task name in dry json. (it was originally implemented without that for this reason, I think?)

Other than that, I'm ok with this

File: cli/internal/runsummary/task_summary.go [hunk 1]
```diff
@@ -51,11 +51,12 @@ func (ts *TaskSummary) cleanForSinglePackage() {
 	for i, dependent := range ts.Dependents {
 		dependents[i] = util.StripPackageName(dependent)
 	}
+	taskID := util.StripPackageName(ts.TaskID)
```
Comment (mehulkar) [inline cli/internal/runsummary/task_summary.go:54]: Can you keep this as `task` though, it's not a taskId anymore

```suggestion
	task := util.StripPackageName(ts.TaskID)
```

File: cli/internal/runsummary/task_summary.go [hunk 2]
```diff
@@ -51,11 +51,12 @@ func (ts *TaskSummary) cleanForSinglePackage() {
...
 
-	ts.Task = util.RootTaskTaskName(ts.TaskID)
+	ts.TaskID = taskID
```
Comment (mehulkar) [inline cli/internal/runsummary/task_summary.go:56]:
```suggestion
	ts.TaskID = task
```

File: cli/internal/runsummary/task_summary.go [hunk 3]
```diff
@@ -51,11 +51,12 @@ func (ts *TaskSummary) cleanForSinglePackage() {
...
+	ts.Task = taskID
```
Comment (mehulkar) [inline cli/internal/runsummary/task_summary.go:57]:
```suggestion
	ts.Task = task
```

**Comment by tknickman** [comment]
> taskId all over the codebase is pkg#task, i'm concerned this exception is confusing

The task ID is a string ID unique to a single task in the array - it shouldn't tell you what the string contains, nor should assumptions be made about it. We provide task, and now package as separate keys which should be sufficient. 

> this duplicates the task name in dry json. (it was originally implemented without that for this reason, I think?)

I'm fine with this, since they have a different meaning (your task name vs the unique ID for the task)

**Comment by mehulkar** [comment]
> The task ID is a string ID unique to a single task in the array

Hmm I don't think that's true, but we can say that now. If you look at some of the ways it's used, it's always assumed to be in the pkg#task format. Eg. look at `GetTaskId()` or any of the other helpers here: https://github.com/vercel/turbo/blob/main/cli/internal/util/task_id.go

**Comment by mehulkar** [comment]
Uniqueness is a key property of `taskID` yes, but I'm just saying it's not the only key property of how taskId is used. As long as this change is restricted to the tail end of a run, we're ok, but if you removed the `//#` part of a `taskId` before the graph is constructed or tasks are executed, I'm pretty sure things would fail

**Pull request merged**